### ISSUE TITLE
[feature request] BehaviorSubject to Signals migration

### ISSUE THREAD

**Issue by ShacharHarshuv** [ngxtension/ngxtension-platform#504]
In the spirit of writing migration scripts to modernize Angular code, I want to propose the following script. Let me know if you think this is something that could fit the library. 

The assumption here that everything that a `BehaviorSubject` is used for, a signal can be used instead, and I think it is possible to write a (relatively) robust script to do that migration. The script will:

- Replace member instantiation of `new BehaviorSubject` with `signal`, while keeping any used generics / inferred types. 
- Rename the member if it ends with `$`
- Replace `myValue$.value` usages to `myValue()`
- Replace any other `myValue$` with `toObservable(myValue)
  - caveat: it might not be in an injection context. A possible solution of it will be duplicating the member in instantiation so that we'll have both `myValue` as a signal and `myValue$` as a stream. In that case, if there is no `$` as a postfix, we'll have to add it.

**Comment by jdegand** [comment]
And convert `.next` to `.set`?

**Comment by ShacharHarshuv** [comment]
Yes, sorry. Added!

**Comment by eneajaho** [comment]
Hi, something is coming 🙌

Let's wait a little bit more 😬

**Comment by ShacharHarshuv** [comment]
You mean you are working on it?

**Comment by eneajaho** [comment]
I've been working on a prototype for myself to do these kinds of migration, but nothing ready yet.

**Comment by ShacharHarshuv** [comment]
Cool! I also thought that a next step could be converting something like that:

```
combineLatest({
  a: toObservable(this.a),
  b: toObservable(this.b),
})
.pipe(
  map(({a, b}) => this.a() + this.b()),
  distinctUntilChanged(), // ... and other operators that don't matter with signals like share etc
);
```
to 
```
computed(() => this.a() + this.b());
```

Might be harder than I think but it feels like it could be possible.

**Comment by kbrilla** [comment]
I would argue that this could be `applyFix` type of migration for VS code, but should not be used with `ng migrate`; Converting all `BehaviorSubject` `signal` may lead to errors, as there may by something along the line of `pipe(pairwise())` somewhere in the code and using `toObservable()` would break this behaviour as instead, beacause of lazy nature of signals instead of `abcd` we would get only `d` delivered to pipe;


https://stackblitz.com/edit/stackblitz-starters-qnm6xx?file=src%2Fmain.ts (look at console to see difrence)
Basicly:
<img width="79" alt="image" src="https://github.com/user-attachments/assets/28322f7d-3b99-43d5-a59f-9fa52cee88c6">
Only BehaviorSubject prints all values, while signal, and toSignal/toObservable prints only last values that were recived in CD cycle because they relay on `effect()`.

**Comment by ShacharHarshuv** [comment]
Yeah they behave differently. Though I feel like in practice if you use a `BehaviorSubject` you **probably** only care about the last value.
